classes/submission/maps/Schema.php

```diff
@@ -165,4 +175,66 @@ protected function getAvailableEditorialDecisions(int $stageId, Submission $subm
 
         return $decisionTypes;
     }
+
+    /**
+     * Get issues associated with submissions. Results are keyed by submission ID.
+     *
+     * @return Enumerable<int, Issue[]>
+     */
+    protected function getSubmissionsIssues(Enumerable $submissions): Enumerable
+    {
+        $submissionIds = $submissions->map(fn (Submission $submission) => $submission->getId())->all();
+        $publications = Repo::publication()->getCollector()->filterBySubmissionIds($submissionIds)->getMany();
+        $issueIds = $publications->map(fn (Publication $publication) => $publication->getData('issueId'))
+            ->unique()
+            ->all();
+
+        $issues = Repo::issue()->getCollector()
+            ->filterByContextIds([$this->context->getId()])
+            ->filterByIssueIds($issueIds)
+            ->getMany();
+
+        $issueIdsGroupedBySubmission = $publications->groupBy(fn (Publication $publication) => $publication->getData('submissionId'))
+            ->map(fn ($entry) => $entry->map(fn (Publication $publication) => $publication->getData('issueId')));
+
+        return $submissions->mapWithKeys(function ($submission) use (&$issues, $publications, $issueIdsGroupedBySubmission) {
+            $submissionIssueIds = $issueIdsGroupedBySubmission->get($submission->getId())->all();
+            return [$submission->getId() => $issues->filter(fn ($issue) => in_array($issue->getId(), $submissionIssueIds))];
+        });
+    }
+
+    /**
+     * Get details about the issue a submission will be published in.
+     */
+    protected function getPropertyIssueToBePublished(Enumerable $publications): ?array
+    {
+        /** @var Publication $latestScheduledPublication */
+        $latestScheduledPublication = $publications
+            ->filter(fn ($publication) => $publication->getData('status') === Submission::STATUS_SCHEDULED)
+            ->sortByDesc(fn (Publication $publication) => $publication->getData('version'))
+            ->first();
+
+        if ($latestScheduledPublication) {
+            $submissionId = $latestScheduledPublication->getData('submissionId');
+            $issueId = $latestScheduledPublication->getData('issueId');
+            $issue = $this->submissionsIssues->get($submissionId, collect())->get($issueId);
+
+            return $issue ? [
+                'id' => $issueId,
+                'label' => $issue->getIssueIdentification()
+            ] : null;
+        }
+
+        return null;
+    }
+
+    /**
+     * Populate class properties specific to OJS.
```

Comment:
```suggestion
     * Populate class properties specific to OJS.
     * @param Enumerable<int, Submission> $submissions Submissions, keyed by submission ID.
```

Reply:
$submissions won't necessarily be keyed by ID, example when this method is called from `map` method in schema, like: `$this->addAppSpecificData(collect([$item]));`, that won't be keyed by submission ID